pkg/api/plugins.go

```diff
@@ -29,21 +29,20 @@ func (hs *HTTPServer) getPluginContext(pluginID string, user *models.SignedInUse
 		return pc, ErrPluginNotFound
 	}
 
-	var jsonData json.RawMessage
-	var decryptedSecureJSONData map[string]string
+	jsonData := json.RawMessage{}
+	decryptedSecureJSONData := map[string]string{}
 	var updated time.Time
 
 	ps, err := hs.getCachedPluginSettings(pluginID, user)
 	if err != nil {
 		if err != models.ErrPluginSettingNotFound {
 			return pc, errutil.Wrap("Failed to get plugin settings", err)
 		}
+	} else {
 		jsonData, err = json.Marshal(ps.JsonData)
```

Comment:
Assuming that the origin of the bug was that if `err == models.ErrPluginSettingNotFound` this line would fail because in that case `ps` is `nil`,
would it be cleaner instead of checking the `err` (by wrapping it in an `else` statement) to check if `ps != nil`?

Another option would be to use switch instead like:
```
	var jsonData json.RawMessage
	var decryptedSecureJSONData map[string]string
	var updated time.Time

	ps, err := hs.getCachedPluginSettings(pluginID, user)
	switch {
	case err != nil:
		switch {
		case err != models.ErrPluginSettingNotFound:
			return pc, errutil.Wrap("Failed to get plugin settings", err)
		default:
			jsonData = json.RawMessage{}
			decryptedSecureJSONData = map[string]string{}
		}
	default:
		jsonData, err = json.Marshal(ps.JsonData)
		if err != nil {
			return pc, errutil.Wrap("Failed to unmarshal plugin json data", err)
		}
		decryptedSecureJSONData = ps.DecryptedValues()
		updated = ps.Updated
	}
```

Reply:
@papagian I think it's better to check if `err` is non-nil as is done here, than to check if `ps` is nil. The latter approach is too lenient; if `err` is nil, `ps` should not be nil. If `err` is not nil, the value of `ps` doesn't matter.